webui/src/__tests__/query.ts

```diff
@@ -7,49 +7,102 @@ it('parses a simple query', () => {
 });
 
 it('parses a query with multiple filters', () => {
-  expect(parse('foo:bar baz:foo-bar')).toEqual({
+  expect(parse(`foo:bar baz:foo-bar`)).toEqual({
     foo: ['bar'],
     baz: ['foo-bar'],
   });
+
+  expect(parse(`label:abc freetext`)).toEqual({
+    label: [`abc`],
+    freetext: [''],
+  });
+
+  expect(parse(`label:abc with "quotes" in freetext`)).toEqual({
+    label: [`abc`],
+    with: [''],
+    quotes: [''],
+    in: [''],
+    freetext: [''],
+  });
 });
 
 it('parses a quoted query', () => {
-  expect(parse('foo:"bar"')).toEqual({
-    foo: ['bar'],
+  expect(parse(`foo:"bar"`)).toEqual({
+    foo: [`"bar"`],
   });
 
-  expect(parse("foo:'bar'")).toEqual({
-    foo: ['bar'],
+  expect(parse(`foo:'bar'`)).toEqual({
+    foo: [`'bar'`],
+  });
+
+  expect(parse(`label:'multi word label'`)).toEqual({
+    label: [`'multi word label'`],
+  });
+
+  expect(parse(`label:"multi word label"`)).toEqual({
+    label: [`"multi word label"`],
+  });
+
+  expect(parse(`label:'multi word label with "nested" quotes'`)).toEqual({
+    label: [`'multi word label with "nested" quotes'`],
   });
 
-  expect(parse('foo:\'bar "nested" quotes\'')).toEqual({
-    foo: ['bar "nested" quotes'],
+  expect(parse(`label:"multi word label with 'nested' quotes"`)).toEqual({
+    label: [`"multi word label with 'nested' quotes"`],
   });
 
-  expect(parse("foo:'escaped\\' quotes'")).toEqual({
-    foo: ["escaped' quotes"],
+  expect(parse(`label:"with:quoated:colon"`)).toEqual({
+    label: [`"with:quoated:colon"`],
+  });
+});
+
+it('should not escape nested quotes', () => {
+  expect(parse(`foo:'do not escape this ->'<- quote'`)).toEqual({
```

Comment:
This behavior is also different in the go parse I believe. The first corresponding quote would terminate the quoted section. It looks like the regex based parser is too greedy and try to match as much as possible?

Reply:
It's fine for now if the JS parser is not matching exactly the go one, just write the correct test and comment the ones that don't pass. That way it's explicit were the mismatch are and we don't have tests that mislead the reader into thinking that the code is correct.

Reply:
Should be fixed. Added also additional test cases to catch greediness.
Let me now if something is missing and before you merge, so that I can get rid of the eslint hotfixes. :-)